File: test-pilot-portal-comprehensive.js
/**
 * Comprehensive Pilot Portal Testing Script
 * Tests all major pilot portal functionality using Puppeteer
 *
 * @author Maurice (Skycruzer)
 * @version 1.0.0
 */

import puppeteer from 'puppeteer'

// ============================================================================
// Configuration
// ============================================================================
const CONFIG = {
  BASE_URL: 'http://localhost:3000',
  HEADLESS: false, // Set to true for CI/CD
  SLOW_MO: 50, // Slow down for visibility
  VIEWPORT: { width: 1920, height: 1080 },
  TIMEOUT: 10000,
  PILOT_CREDENTIALS: {
    email: process.env.PILOT_EMAIL || 'mrondeau@airniugini.com.pg',
    password: process.env.PILOT_PASSWORD || 'Lemakot@1972',
  },
}

// ============================================================================
// Utility Functions
// ============================================================================
const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms))

const logSection = (title) => {
  console.log('\n' + '='.repeat(80))
  console.log(`  ${title}`)
  console.log('='.repeat(80))
}

const logTest = (testNumber, description) => {
  console.log(`\n🧪 TEST ${testNumber}: ${description}`)
}

const logSuccess = (message) => console.log(`  ✅ ${message}`)
const logError = (message) => console.log(`  ❌ ${message}`)
const logInfo = (message) => console.log(`  ℹ️  ${message}`)
const logWarning = (message) => console.log(`  ⚠️  ${message}`)

// ============================================================================
// Test Results Tracker
// ============================================================================
class TestResults {
  constructor() {
    this.passed = 0
    this.failed = 0
    this.skipped = 0
    this.tests = []
  }

  pass(testName) {
    this.passed++
    this.tests.push({ name: testName, status: 'PASSED' })
    logSuccess(`${testName} - PASSED`)
  }

  fail(testName, error = '') {
    this.failed++
    this.tests.push({ name: testName, status: 'FAILED', error })
    logError(`${testName} - FAILED${error ? ': ' + error : ''}`)
  }

  skip(testName, reason = '') {
    this.skipped++
    this.tests.push({ name: testName, status: 'SKIPPED', reason })
    logWarning(`${testName} - SKIPPED${reason ? ': ' + reason : ''}`)
  }

  getSummary() {
    const total = this.passed + this.failed + this.skipped
    const successRate =
      this.passed + this.failed > 0
        ? ((this.passed / (this.passed + this.failed)) * 100).toFixed(1)
        : 0
    return {
      total,
      passed: this.passed,
      failed: this.failed,
      skipped: this.skipped,
      successRate: `${successRate}%`,
    }
  }

  printSummary() {
    const summary = this.getSummary()
    logSection('📊 TEST SUMMARY')
    console.log(`Total Tests: ${summary.total}`)
    console.log(`✅ Passed: ${summary.passed}`)
    console.log(`❌ Failed: ${summary.failed}`)
    console.log(`⏭️  Skipped: ${summary.skipped}`)
    console.log(`📈 Success Rate: ${summary.successRate}`)
    console.log('='.repeat(80))

    if (this.failed === 0) {
      console.log('\n🎉 ALL TESTS PASSED! Pilot portal is working perfectly!')
    } else {
      console.log(`\n⚠️  ${this.failed} test(s) failed. Review errors above.`)
    }
  }
}

// ============================================================================
// API Response Interceptor
// ============================================================================
class APIMonitor {
  constructor(page) {
    this.responses = {}
    this.errors = []

    page.on('response', async (response) => {
      const url = response.url()
      if (url.includes('/api/portal/')) {
        const apiName = url.split('/api/portal/')[1].split('?')[0]
        this.responses[apiName] = {
          status: response.status(),
          ok: response.ok(),
          url,
          timestamp: new Date().toISOString(),
        }
      }
    })

    page.on('console', (msg) => {
      if (msg.type() === 'error') {
        this.errors.push(msg.text())
      }
    })
  }

  getResponse(apiName) {
    return this.responses[apiName]
  }

  hasError(apiName) {
    const response = this.responses[apiName]
    return response && !response.ok
  }

  printSummary() {
    logSection('📡 API RESPONSE SUMMARY')
    if (Object.keys(this.responses).length === 0) {
      logWarning('No API responses captured')
      return
    }

    Object.entries(this.responses).forEach(([api, response]) => {
      const icon = response.ok ? '✅' : '❌'
      console.log(`${icon} /api/portal/${api}: ${response.status} ${response.ok ? 'OK' : 'FAILED'}`)
    })

    if (this.errors.length > 0) {
      console.log('\n⚠️  Console Errors Detected:')
      this.errors.slice(0, 5).forEach((err) => console.log(`  - ${err}`))
      if (this.errors.length > 5) {
        console.log(`  ... and ${this.errors.length - 5} more errors`)
      }
    }
  }
}

// ============================================================================
// Screenshot Helper
// ============================================================================
async function takeScreenshot(page, name) {
  try {
    const filename = `./test-screenshots/pilot-portal-${name}-${Date.now()}.png`
    await page.screenshot({ path: filename, fullPage: true })
    logInfo(`Screenshot saved: ${filename}`)
  } catch (error) {
    logWarning(`Failed to take screenshot: ${error.message}`)
  }
}

// ============================================================================
// Pilot Portal Test Suite
// ============================================================================
async function testPilotPortal() {
  logSection('🚀 STARTING PILOT PORTAL COMPREHENSIVE TESTING')

  const results = new TestResults()
  let browser, page, apiMonitor

  try {
    // ========================================================================
    // Setup Browser
    // ========================================================================
    logInfo('Launching browser...')
    browser = await puppeteer.launch({
      headless: CONFIG.HEADLESS,
      slowMo: CONFIG.SLOW_MO,
      args: [
        `--window-size=${CONFIG.VIEWPORT.width},${CONFIG.VIEWPORT.height}`,
        '--no-sandbox',
        '--disable-setuid-sandbox',
      ],
    })

    page = await browser.newPage()
    await page.setViewport(CONFIG.VIEWPORT)
    apiMonitor = new APIMonitor(page)
    logSuccess('Browser launched successfully')

    // ========================================================================
    // TEST 1: Navigate to Pilot Portal Login
    // ========================================================================
    logTest(1, 'Navigate to Pilot Portal Login Page')
    await page.goto(`${CONFIG.BASE_URL}/portal/login`, {
      waitUntil: 'networkidle2',
      timeout: CONFIG.TIMEOUT,
    })

    const loginPageTitle = await page.title()
    if (loginPageTitle && page.url().includes('/portal/login')) {
      results.pass('Pilot login page loads')
    } else {
      results.fail('Pilot login page loads', 'Page did not load correctly')
      await takeScreenshot(page, 'login-page-error')
    }
    await sleep(1500)

    // ========================================================================
    // TEST 2: Pilot Authentication
    // ========================================================================
    logTest(2, 'Pilot Login Authentication')

    try {
      // Wait for and fill email
      await page.waitForSelector('input[type="email"]', { timeout: 5000 })
      await page.type('input[type="email"]', CONFIG.PILOT_CREDENTIALS.email, { delay: 50 })
      logInfo('Email entered')

      // Wait for and fill password
      await page.waitForSelector('input[type="password"]', { timeout: 5000 })
      await page.type('input[type="password"]', CONFIG.PILOT_CREDENTIALS.password, { delay: 50 })
      logInfo('Password entered')

      // Click login button
      await page.click('button[type="submit"]')
      logInfo('Login button clicked')

      // Wait for navigation
      await page.waitForNavigation({ waitUntil: 'networkidle2', timeout: CONFIG.TIMEOUT })

      const currentUrl = page.url()
      const cookies = await page.cookies()

      if (currentUrl.includes('/portal/dashboard')) {
        results.pass('Pilot login successful')
        logInfo(`Session cookies set: ${cookies.length} cookies`)
      } else {
        results.fail('Pilot login successful', `Redirected to: ${currentUrl}`)
        await takeScreenshot(page, 'login-failed')
      }
    } catch (error) {
      results.fail('Pilot login successful', error.message)
      await takeScreenshot(page, 'login-error')
    }
    await sleep(2000)

    // ========================================================================
    // TEST 3: Dashboard Page
    // ========================================================================
    logTest(3, 'Pilot Dashboard Page')

    try {
      const dashboardTitle = await page.title()
      const hasContent = await page.evaluate(() => {
        const headings = document.querySelectorAll('h1, h2, h3')
        return headings.length > 0
      })

      if (hasContent && page.url().includes('/portal/dashboard')) {
        results.pass('Dashboard page displays correctly')
      } else {
        results.fail('Dashboard page displays correctly', 'Content not found')
        await takeScreenshot(page, 'dashboard-error')
      }
    } catch (error) {
      results.fail('Dashboard page displays correctly', error.message)
    }
    await sleep(2000)

    // ========================================================================
    // TEST 4: Profile Page
    // ========================================================================
    logTest(4, 'Profile Page')

    try {
      await page.goto(`${CONFIG.BASE_URL}/portal/profile`, {
        waitUntil: 'networkidle2',
        timeout: CONFIG.TIMEOUT,
      })
      await sleep(2000)

      const profileResponse = apiMonitor.getResponse('profile')
      if (profileResponse && profileResponse.ok) {
        results.pass('Profile page loads with pilot data')
        logInfo(`API Status: ${profileResponse.status}`)
      } else {
        results.fail(
          'Profile page loads with pilot data',
          `API status: ${profileResponse?.status || 'N/A'}`
        )
      }
    } catch (error) {
      results.fail('Profile page loads with pilot data', error.message)
    }

    // ========================================================================
    // TEST 5: Certifications Page
    // ========================================================================
    logTest(5, 'Certifications Page')

    try {
      await page.goto(`${CONFIG.BASE_URL}/portal/certifications`, {
        waitUntil: 'networkidle2',
        timeout: CONFIG.TIMEOUT,
      })
      await sleep(2000)

      const certsResponse = apiMonitor.getResponse('certifications')
      if (certsResponse && certsResponse.ok) {
        results.pass('Certifications page loads')
        logInfo(`API Status: ${certsResponse.status}`)
      } else {
        results.fail('Certifications page loads', `API status: ${certsResponse?.status || 'N/A'}`)
      }
    } catch (error) {
      results.fail('Certifications page loads', error.message)
    }

    // ========================================================================
    // TEST 6: Leave Requests Page
    // ========================================================================
    logTest(6, 'Leave Requests Page')

    try {
      await page.goto(`${CONFIG.BASE_URL}/portal/leave-requests`, {
        waitUntil: 'networkidle2',
        timeout: CONFIG.TIMEOUT,
      })
      await sleep(2000)

      const leaveResponse = apiMonitor.getResponse('leave-requests')
      if (leaveResponse && leaveResponse.ok) {
        results.pass('Leave requests page loads')
        logInfo(`API Status: ${leaveResponse.status}`)
      } else {
        results.fail('Leave requests page loads', `API status: ${leaveResponse?.status || 'N/A'}`)
      }
    } catch (error) {
      results.fail('Leave requests page loads', error.message)
    }

    // ========================================================================
    // TEST 7: Flight Requests Page
    // ========================================================================
    logTest(7, 'Flight Requests Page')

    try {
      await page.goto(`${CONFIG.BASE_URL}/portal/flight-requests`, {
        waitUntil: 'networkidle2',
        timeout: CONFIG.TIMEOUT,
      })
      await sleep(2000)

      const flightResponse = apiMonitor.getResponse('flight-requests')
      if (flightResponse && flightResponse.ok) {
        results.pass('Flight requests page loads')
        logInfo(`API Status: ${flightResponse.status}`)
      } else {
        results.fail('Flight requests page loads', `API status: ${flightResponse?.status || 'N/A'}`)
      }
    } catch (error) {
      results.fail('Flight requests page loads', error.message)
    }

    // ========================================================================
    // TEST 8: Notifications Page
    // ========================================================================
    logTest(8, 'Notifications Page')

    try {
      await page.goto(`${CONFIG.BASE_URL}/portal/notifications`, {
        waitUntil: 'networkidle2',
        timeout: CONFIG.TIMEOUT,
      })
      await sleep(2000)

      const notifResponse = apiMonitor.getResponse('notifications')
      if (notifResponse && notifResponse.ok) {
        results.pass('Notifications page loads')
        logInfo(`API Status: ${notifResponse.status}`)
      } else {
        results.fail('Notifications page loads', `API status: ${notifResponse?.status || 'N/A'}`)
      }
    } catch (error) {
      results.fail('Notifications page loads', error.message)
    }

    // ========================================================================
    // TEST 9: Feedback Page
    // ========================================================================
    logTest(9, 'Feedback/Support Page')

    try {
      await page.goto(`${CONFIG.BASE_URL}/portal/feedback`, {
        waitUntil: 'networkidle2',
        timeout: CONFIG.TIMEOUT,
      })
      await sleep(2000)

      if (page.url().includes('/portal/feedback')) {
        results.pass('Feedback page loads')
      } else {
        results.fail('Feedback page loads', 'Page not found')
      }
    } catch (error) {
      results.fail('Feedback page loads', error.message)
    }

    // ========================================================================
    // TEST 10: Navigation Menu
    // ========================================================================
    logTest(10, 'Navigation Menu Functionality')

    try {
      await page.goto(`${CONFIG.BASE_URL}/portal/dashboard`, {
        waitUntil: 'networkidle2',
      })
      await sleep(1000)

      const navLinks = await page.evaluate(() => {
        const links = Array.from(document.querySelectorAll('nav a, aside a'))
        return links
          .map((link) => ({
            text: link.textContent.trim(),
            href: link.getAttribute('href'),
          }))
          .filter((link) => link.href && link.href.includes('/portal/'))
      })

      if (navLinks.length >= 5) {
        results.pass('Navigation menu functional')
        logInfo(`Found ${navLinks.length} navigation links`)
        navLinks.forEach((link) => {
          console.log(`    - ${link.text}: ${link.href}`)
        })
      } else {
        results.fail('Navigation menu functional', `Only ${navLinks.length} links found`)
      }
    } catch (error) {
      results.fail('Navigation menu functional', error.message)
    }

    // ========================================================================
    // TEST 11: Create New Leave Request (Navigation)
    // ========================================================================
    logTest(11, 'Create New Leave Request (Navigation)')

    try {
      await page.goto(`${CONFIG.BASE_URL}/portal/leave-requests/new`, {
        waitUntil: 'networkidle2',
        timeout: CONFIG.TIMEOUT,
      })
      await sleep(1500)

      const hasForm = await page.evaluate(() => {
        return document.querySelector('form') !== null
      })

      if (hasForm && page.url().includes('/leave-requests/new')) {
        results.pass('New leave request form accessible')
      } else {
        results.fail('New leave request form accessible', 'Form not found')
      }
    } catch (error) {
      results.fail('New leave request form accessible', error.message)
    }

    // ========================================================================
    // TEST 12: Create New Flight Request (Navigation)
    // ========================================================================
    logTest(12, 'Create New Flight Request (Navigation)')

    try {
      await page.goto(`${CONFIG.BASE_URL}/portal/flight-requests/new`, {
        waitUntil: 'networkidle2',
        timeout: CONFIG.TIMEOUT,
      })
      await sleep(1500)

      const hasForm = await page.evaluate(() => {
        return document.querySelector('form') !== null
      })

      if (hasForm && page.url().includes('/flight-requests/new')) {
        results.pass('New flight request form accessible')
      } else {
        results.fail('New flight request form accessible', 'Form not found')
      }
    } catch (error) {
      results.fail('New flight request form accessible', error.message)
    }

    // ========================================================================
    // TEST 13: Mobile Responsive Design
    // ========================================================================
    logTest(13, 'Mobile Responsive Design')

    try {
      // Test iPhone SE viewport
      await page.setViewport({ width: 375, height: 667 })
      await page.goto(`${CONFIG.BASE_URL}/portal/dashboard`, {
        waitUntil: 'networkidle2',
      })
      await sleep(1500)

      const isMobileResponsive = await page.evaluate(() => {
        const width = window.innerWidth
        const hasOverflow = document.body.scrollWidth > window.innerWidth
        return width === 375 && !hasOverflow
      })

      if (isMobileResponsive) {
        results.pass('Mobile responsive design works')
      } else {
        results.fail('Mobile responsive design works', 'Horizontal overflow detected')
        await takeScreenshot(page, 'mobile-view')
      }

      // Reset viewport
      await page.setViewport(CONFIG.VIEWPORT)
    } catch (error) {
      results.fail('Mobile responsive design works', error.message)
    }

    // ========================================================================
    // TEST 14: Session Persistence
    // ========================================================================
    logTest(14, 'Session Persistence')

    try {
      // Navigate away and back
      await page.goto(`${CONFIG.BASE_URL}`, { waitUntil: 'networkidle2' })
      await sleep(1000)
      await page.goto(`${CONFIG.BASE_URL}/portal/dashboard`, { waitUntil: 'networkidle2' })
      await sleep(1500)

      const stillAuthenticated = page.url().includes('/portal/dashboard')
      if (stillAuthenticated) {
        results.pass('Session persists across navigation')
      } else {
        results.fail('Session persists across navigation', 'Session lost')
      }
    } catch (error) {
      results.fail('Session persists across navigation', error.message)
    }

    // ========================================================================
    // TEST 15: Logout Functionality
    // ========================================================================
    logTest(15, 'Logout Functionality')

    try {
      // Look for logout button
      const logoutButton = await page.evaluate(() => {
        const buttons = Array.from(document.querySelectorAll('button, a'))
        const logoutBtn = buttons.find(
          (btn) =>
            btn.textContent.toLowerCase().includes('logout') ||
            btn.textContent.toLowerCase().includes('sign out')
        )
        return logoutBtn !== undefined
      })

      if (logoutButton) {
        results.pass('Logout button available')
        logInfo('Logout button found in navigation')
      } else {
        results.skip('Logout button available', 'Logout button not found in current implementation')
      }
    } catch (error) {
      results.skip('Logout button available', error.message)
    }

    // ========================================================================
    // Print Results
    // ========================================================================
    console.log('\n')
    apiMonitor.printSummary()
    results.printSummary()
  } catch (error) {
    console.error('\n❌ CRITICAL ERROR:', error.message)
    console.error(error.stack)
    if (page) {
      await takeScreenshot(page, 'critical-error')
    }
  } finally {
    if (browser) {
      logInfo('Keeping browser open for 5 seconds...')
      await sleep(5000)
      await browser.close()
      logSuccess('Browser closed')
    }

    // Exit with appropriate code
    process.exit(results.failed > 0 ? 1 : 0)
  }
}

// ============================================================================
// Run Tests
// ============================================================================
testPilotPortal().catch(console.error)

File: test_pilot_portal_comprehensive.py
#!/usr/bin/env python3
"""
Comprehensive Pilot Portal Test
Tests login and all pilot portal features in detail
"""

from playwright.sync_api import sync_playwright
import sys
import time

# Pilot Portal Credentials
PILOT_EMAIL = "mrondeau@airniugini.com.pg"
PILOT_PASSWORD = "mron2393"

def test_pilot_portal_comprehensive():
    """Comprehensive test of Pilot Portal"""

    results = {
        "passed": [],
        "failed": [],
        "screenshots": [],
        "features": []
    }

    with sync_playwright() as p:
        browser = p.chromium.launch(headless=False)
        context = browser.new_context()
        page = context.new_page()

        try:
            print("\n" + "="*70)
            print("✈️  COMPREHENSIVE PILOT PORTAL TEST")
            print("="*70)

            # ===================================================================
            # TEST 1: LOGIN
            # ===================================================================
            print("\n🧪 TEST 1: Pilot Portal Login")
            print("-" * 70)

            page.goto('http://localhost:3003/portal/login')
            page.wait_for_load_state('networkidle')
            time.sleep(2)

            page.screenshot(path='/tmp/pilot_comprehensive_01_login.png', full_page=True)
            results["screenshots"].append('/tmp/pilot_comprehensive_01_login.png')

            # Fill credentials
            page.fill('input[type="email"]', PILOT_EMAIL)
            page.fill('input[type="password"]', PILOT_PASSWORD)

            print(f"   📧 Email: {PILOT_EMAIL}")
            print("   🔒 Password: ********")

            # Submit
            page.click('button[type="submit"]')
            page.wait_for_load_state('networkidle', timeout=10000)
            time.sleep(3)

            page.screenshot(path='/tmp/pilot_comprehensive_02_after_login.png', full_page=True)
            results["screenshots"].append('/tmp/pilot_comprehensive_02_after_login.png')

            current_url = page.url
            print(f"   🌐 Current URL: {current_url}")

            if '/portal' in current_url and '/login' not in current_url:
                print("   ✅ Login SUCCESSFUL - Redirected to portal")
                results["passed"].append("Login successful")
            else:
                print("   ❌ Login FAILED - Still on login page")
                try:
                    error = page.locator('[class*="error"]').first.text_content(timeout=1000)
                    print(f"   ❌ Error: {error}")
                    results["failed"].append(f"Login failed: {error}")
                except:
                    results["failed"].append("Login failed - unknown error")
                # Stop if login failed
                return 1

            # ===================================================================
            # TEST 2: DASHBOARD
            # ===================================================================
            print("\n🧪 TEST 2: Dashboard - Overview & Metrics")
            print("-" * 70)

            page.goto('http://localhost:3003/portal/dashboard')
            page.wait_for_load_state('networkidle')
            time.sleep(2)

            page.screenshot(path='/tmp/pilot_comprehensive_03_dashboard.png', full_page=True)
            results["screenshots"].append('/tmp/pilot_comprehensive_03_dashboard.png')

            # Check for dashboard elements
            try:
                headings = page.locator('h1, h2, h3').all()
                cards = page.locator('[class*="card"]').all()
                buttons = page.locator('button').all()

                print(f"   📊 Found {len(headings)} headings")
                print(f"   📦 Found {len(cards)} cards")
                print(f"   🔘 Found {len(buttons)} buttons")

                if len(headings) > 0:
                    print(f"   📌 Main heading: {headings[0].text_content()}")
                    results["passed"].append("Dashboard loaded with content")
                    results["features"].append(f"Dashboard: {len(cards)} cards")
            except Exception as e:
                print(f"   ⚠️  Dashboard check error: {e}")

            # ===================================================================
            # TEST 3: MY PROFILE
            # ===================================================================
            print("\n🧪 TEST 3: My Profile - Personal Information")
            print("-" * 70)

            page.goto('http://localhost:3003/portal/profile')
            page.wait_for_load_state('networkidle')
            time.sleep(2)

            page.screenshot(path='/tmp/pilot_comprehensive_04_profile.png', full_page=True)
            results["screenshots"].append('/tmp/pilot_comprehensive_04_profile.png')

            # Check for profile fields
            try:
                labels = page.locator('label').all()
                inputs = page.locator('input').all()

                print(f"   👤 Found {len(labels)} labels")
                print(f"   📝 Found {len(inputs)} input fields")

                # Try to read pilot name
                try:
                    name_element = page.locator('h1, h2').first
                    pilot_name = name_element.text_content()
                    print(f"   ✈️  Pilot: {pilot_name}")
                    results["features"].append(f"Profile: {pilot_name}")
                except:
                    pass

                results["passed"].append("Profile page loaded")
            except Exception as e:
                print(f"   ⚠️  Profile check error: {e}")

            # ===================================================================
            # TEST 4: MY CERTIFICATIONS
            # ===================================================================
            print("\n🧪 TEST 4: My Certifications - Training Records")
            print("-" * 70)

            page.goto('http://localhost:3003/portal/certifications')
            page.wait_for_load_state('networkidle')
            time.sleep(2)

            page.screenshot(path='/tmp/pilot_comprehensive_05_certifications.png', full_page=True)
            results["screenshots"].append('/tmp/pilot_comprehensive_05_certifications.png')

            # Check for certification data
            try:
                rows = page.locator('tr').all()
                tables = page.locator('table').all()

                print(f"   📋 Found {len(tables)} tables")
                print(f"   📊 Found {len(rows)} table rows")

                if len(rows) > 1:
                    print(f"   ✅ Certifications: {len(rows) - 1} records")
                    results["features"].append(f"Certifications: {len(rows)-1} records")
                else:
                    print("   ℹ️  No certification data visible")

                results["passed"].append("Certifications page loaded")
            except Exception as e:
                print(f"   ⚠️  Certifications check error: {e}")

            # ===================================================================
            # TEST 5: LEAVE REQUESTS
            # ===================================================================
            print("\n🧪 TEST 5: My Leave Requests - Time Off Management")
            print("-" * 70)

            page.goto('http://localhost:3003/portal/leave-requests')
            page.wait_for_load_state('networkidle')
            time.sleep(2)

            page.screenshot(path='/tmp/pilot_comprehensive_06_leave.png', full_page=True)
            results["screenshots"].append('/tmp/pilot_comprehensive_06_leave.png')

            # Check for leave request features
            try:
                tables = page.locator('table').all()
                buttons = page.locator('button').all()
                forms = page.locator('form').all()

                print(f"   📋 Found {len(tables)} tables")
                print(f"   🔘 Found {len(buttons)} buttons")
                print(f"   📝 Found {len(forms)} forms")

                # Check for "New Request" or "Submit" button
                try:
                    submit_btn = page.locator('button:has-text("Submit"), button:has-text("New"), button:has-text("Create")').first
                    print(f"   ✅ Action button found: Submit/Create available")
                    results["features"].append("Leave Requests: Can submit new requests")
                except:
                    pass

                results["passed"].append("Leave requests page loaded")
            except Exception as e:
                print(f"   ⚠️  Leave requests check error: {e}")

            # ===================================================================
            # TEST 6: FLIGHT REQUESTS
            # ===================================================================
            print("\n🧪 TEST 6: My Flight Requests - Flight Scheduling")
            print("-" * 70)

            page.goto('http://localhost:3003/portal/flight-requests')
            page.wait_for_load_state('networkidle')
            time.sleep(2)

            page.screenshot(path='/tmp/pilot_comprehensive_07_flights.png', full_page=True)
            results["screenshots"].append('/tmp/pilot_comprehensive_07_flights.png')

            # Check for flight request features
            try:
                tables = page.locator('table').all()
                buttons = page.locator('button').all()

                print(f"   📋 Found {len(tables)} tables")
                print(f"   🔘 Found {len(buttons)} buttons")

                # Check for submit capability
                try:
                    submit_btn = page.locator('button:has-text("Submit"), button:has-text("New"), button:has-text("Create")').first
                    print(f"   ✅ Action button found: Can submit flight requests")
                    results["features"].append("Flight Requests: Submission enabled")
                except:
                    pass

                results["passed"].append("Flight requests page loaded")
            except Exception as e:
                print(f"   ⚠️  Flight requests check error: {e}")

            # ===================================================================
            # TEST 7: NOTIFICATIONS (if exists)
            # ===================================================================
            print("\n🧪 TEST 7: Notifications - Alerts & Updates")
            print("-" * 70)

            try:
                page.goto('http://localhost:3003/portal/notifications')
                page.wait_for_load_state('networkidle', timeout=5000)
                time.sleep(2)

                page.screenshot(path='/tmp/pilot_comprehensive_08_notifications.png', full_page=True)
                results["screenshots"].append('/tmp/pilot_comprehensive_08_notifications.png')

                print("   ✅ Notifications page exists")
                results["passed"].append("Notifications page loaded")
            except:
                print("   ℹ️  Notifications page not available (may not exist)")

            # ===================================================================
            # TEST 8: SETTINGS/ACCOUNT (if exists)
            # ===================================================================
            print("\n🧪 TEST 8: Account Settings")
            print("-" * 70)

            try:
                page.goto('http://localhost:3003/portal/settings')
                page.wait_for_load_state('networkidle', timeout=5000)
                time.sleep(2)

                page.screenshot(path='/tmp/pilot_comprehensive_09_settings.png', full_page=True)
                results["screenshots"].append('/tmp/pilot_comprehensive_09_settings.png')

                print("   ✅ Settings page exists")
                results["passed"].append("Settings page loaded")
            except:
                print("   ℹ️  Settings page not available (may not exist)")

            # ===================================================================
            # FINAL: LOGOUT
            # ===================================================================
            print("\n🧪 FINAL: Logout Test")
            print("-" * 70)

            try:
                # Look for logout button/link
                logout_btn = page.locator('button:has-text("Logout"), a:has-text("Logout"), button:has-text("Sign Out")').first
                logout_btn.click()
                time.sleep(2)

                page.screenshot(path='/tmp/pilot_comprehensive_10_logout.png', full_page=True)
                results["screenshots"].append('/tmp/pilot_comprehensive_10_logout.png')

                if '/login' in page.url:
                    print("   ✅ Logout successful - redirected to login")
                    results["passed"].append("Logout successful")
                else:
                    print(f"   ⚠️  Logout attempted, current URL: {page.url}")
            except Exception as e:
                print(f"   ℹ️  Logout button not found: {e}")

        except Exception as e:
            print(f"\n❌ Critical error: {e}")
            results["failed"].append(f"Critical error: {e}")
            page.screenshot(path='/tmp/pilot_comprehensive_error.png', full_page=True)
            results["screenshots"].append('/tmp/pilot_comprehensive_error.png')

        finally:
            print("\n⏳ Keeping browser open for 3 seconds for review...")
            time.sleep(3)
            browser.close()

    # ===================================================================
    # SUMMARY
    # ===================================================================
    print("\n" + "="*70)
    print("📊 COMPREHENSIVE TEST SUMMARY")
    print("="*70)

    print(f"\n✅ PASSED: {len(results['passed'])}")
    for test in results["passed"]:
        print(f"   • {test}")

    if results["failed"]:
        print(f"\n❌ FAILED: {len(results['failed'])}")
        for test in results["failed"]:
            print(f"   • {test}")

    if results["features"]:
        print(f"\n🎯 FEATURES VERIFIED: {len(results['features'])}")
        for feature in results["features"]:
            print(f"   • {feature}")

    print(f"\n📸 SCREENSHOTS: {len(results['screenshots'])}")
    for screenshot in results["screenshots"]:
        print(f"   • {screenshot}")

    print("\n" + "="*70)

    return 0 if len(results["failed"]) == 0 else 1

if __name__ == "__main__":
    print("\n🚀 Starting Comprehensive Pilot Portal Testing...")
    print(f"📧 Email: {PILOT_EMAIL}")
    print(f"🔒 Password: ********")
    sys.exit(test_pilot_portal_comprehensive())
